### ISSUE TITLE
Is it possible to add custom proxy?

### ISSUE THREAD

**Issue by OmiiiDev** [Kav-K/GPTDiscord#461]
Can I use this with a custom OpenAI api proxy URL?

**Comment by AndrewsPanda** [comment]
It would be good. It would allow for more llm's from HF or openrouter to be used with this bot, making it a lot more versatile. 

I've mostly been used this bot in that way by modifying the urls in, /models/opennai_model.py, into f-string values and having it pull 'https://api.openai.com/v1'   from .env. 

The langchain requests also need to be updated. 

I know how to do that much but there are other functions that generally break. 

@Kav-K im happy to do a PR request for the changes i know how to do. 

Its basically the same request as my ticket below. 

https://github.com/Kav-K/GPTDiscord/issues/454

**Comment by Kav-K** [comment]
@AndrewsPanda Yeah sounds great! If you want to get started on a PR with what you can do, I will build on top of it and finish/polish it up!

This is definitely something that I've wanted to implement, added this to the roadmap and prioritized it.
